[Rust] Inline error reporting can be improved using --message-format short

### Expected behavior

`rustc` and `cargo` support a `--message-format` argument that can output errors in a more compact format than the default multi-line human-readable one. Here's an example of how a `--message-format short` error looks:

```
src/lib.rs:625:24: error[E0599]: no method named `clockz` found for reference `&Runtime<&dyn ErasedEmitter + Send + Sync, &dyn ErasedFilter + Send + Sync, &..., ..., ...>` in the current scope
```

This format is nicely compatible with the build system capture group ordering. When matched with a regex:

```
"file_regex": "^([^:]*):([0-9]*):([0-9]*):\\s*(.*)"
```

it can report errors in the editor like so:

![after](https://github.com/sublimehq/Packages/assets/6721458/6268fc6a-caaf-4679-8e9e-6c31c510c735)


### Actual behavior

Compared to the original format, we get a less useful error in the editor, but a more useful error in the build output:

![before](https://github.com/sublimehq/Packages/assets/6721458/d68f981b-627e-4898-9b1b-b982996e6caf)

I personally prefer the `--message-format short` version, but I don't think it's universally better if you're not already familiar with Rust to interpret its errors. I thought I'd open up an issue for it just to demonstrate how it could be different and see if that's preferable to anyone else.

### Steps to reproduce

1. Write a Rust program that includes an error
2. Build using the default Rust plugin and observe the error in the editor

--- a/Rust/Cargo.sublime-build
+++ b/Rust/Cargo.sublime-build
@@ -12,8 +12,18 @@
             "name": "Run"
         },
         {
+            "cmd": ["cargo", "run", "--message-format", "short"],
+            "file_regex": "^([^:]*):([0-9]*):([0-9]*):\\s*(.*)",
+            "name": "Run (Short)"
+        },
+        {
             "cmd": ["cargo", "test"],
             "name": "Test"
+        },
+        {
+            "cmd": ["cargo", "test", "--message-format", "short"],
+            "file_regex": "^([^:]*):([0-9]*):([0-9]*):\\s*(.*)",
+            "name": "Test (Short)"
         },
         {
             "cmd": ["cargo", "bench"],
@@ -26,6 +36,20 @@
         {
             "cmd": ["cargo", "clippy"],
             "name": "Clippy"
+        },
+        {
+            "cmd": ["cargo", "clippy", "--message-format", "short"],
+            "file_regex": "^([^:]*):([0-9]*):([0-9]*):\\s*(.*)",
+            "name": "Clippy (Short)"
+        },
+        {
+            "cmd": ["cargo", "check", "--all-targets"],
+            "name": "Check"
+        },
+        {
+            "cmd": ["cargo", "check", "--all-targets", "--message-format", "short"],
+            "file_regex": "^([^:]*):([0-9]*):([0-9]*):\\s*(.*)",
+            "name": "Check (Short)"
         },
     ]
 }

